src/QMCHamiltonians/BareKineticEnergy.h

```diff
@@ -20,6 +20,7 @@
 #include "Particle/WalkerSetRef.h"
 #include "QMCHamiltonians/QMCHamiltonianBase.h"
 #include "ParticleBase/ParticleAttribOps.h"
+#include "QMCWaveFunctions/TrialWaveFunction.h"
```

Comment:
Is this line really needed?

Reply:
Actually, I don't think it is.

Reply:
Rhea seems to think it is necessary.  Adding it back in.

Reply:
The major drawback of including TrialWaveFunction.h is that it will be propagated.
QMCHamiltonianBase.h intentionally use "class TrialWaveFunction;" to avoid the header inclusion and the derived class cpp includes TrialWaveFunction.h. In your case, the class is a template and there is no cpp. So you have to include the header file.
I believe BareKineticEnergy.h itself will not be propagated too far. So the inclusion is fine.